sdk/storage/Azure.Storage.Files.DataLake/src/Sas/DataLakeSasBuilder.cs

```diff
@@ -342,18 +342,33 @@ private static string GetCanonicalName(string account, string fileSystemName, st
         /// Ensure the <see cref="DataLakeSasBuilder"/>'s properties are in a
         /// consistent state.
         /// </summary>
-        private void EnsureState()
```

Comment:
>EnsureState [](start = 21, length = 11)

It might be helpful to add documentation when a user sets identifier or expiry time and permissions indicating what this ensure state method does